### DISCUSSION TITLE
error in the load energy

### DISCUSSION THREAD

**Discussion by nebbiadigiorno** [slipx06/sunsynk-power-flow-card#261] [Q&A]
hi, i noticed an error in the total power use by the house (se the screenshot).
<img width="1187" alt="Screenshot 2024-02-19 alle 11 18 13" src="https://github.com/slipx06/sunsynk-power-flow-card/assets/59627032/76028ed5-f2c5-4b1c-b537-1a9ccc97f018">

this is the code for the card that i am using:

`type: custom:sunsynk-power-flow-card
cardstyle: lite
show_solar: true
battery:
  energy: 20000
  shutdown_soc: 20
  show_daily: true
  invert_power: true
  show_absolute: false
  hide_soc: false
  show_remaining_energy: true
  auto_scale: true
  dynamic_colour: true
  linear_gradient: true
  max_power: 10000
solar:
  show_daily: true
  mppts: 2
  auto_scale: true
  dynamic_colour: true
  max_power: 10000
  display_mode: 2
load:
  show_daily: true
  auto_scale: true
  dynamic_colour: true
  dynamic_icon: true
  max_power: 25000
  show_aux: false
  additional_loads: 1
  load1_name: bmw i4
  load1_icon: mdi:ev-station
grid:
  show_daily_buy: true
  show_daily_sell: true
  show_nonessential: false
  auto_scale: true
  invert_grid: false
  max_power: 25000
  energy_cost_decimals: 0
entities:
  use_timer_248: switch.sunsynk_toggle_system_timer
  inverter_voltage_154: sensor.solarman_voltage_phase_r
  inverter_current_164: sensor.solarman_current_output_r
  inverter_power_175: sensor.solarman_activepower_output_total
  grid_connected_status_194: binary_sensor.sunsynk_grid_connected_status
  inverter_status_59: sensor.solarman_inverter_status
  day_battery_charge_70: sensor.solarman_daily_battery_charge
  day_battery_discharge_71: sensor.solarman_daily_battery_discharge
  battery_voltage_183: sensor.solarman_battery_1_voltage
  battery_soc_184: sensor.solarman_battery_1_soc
  battery_power_190: sensor.solarman_battery_1_power
  battery_current_191: sensor.solarman_battery_1_current
  grid_power_169: sensor.power_total_istant
  day_grid_import_76: sensor.solarman_daily_energy_bought
  day_grid_export_77: sensor.solarman_daily_energy_sold
  grid_ct_power_172: sensor.phase_a_power
  day_load_energy_84: sensor.solarman_daily_load_consumption
  nonessential_power: none
  day_pv_energy_108: sensor.solarman_daily_pv_generation
  pv1_power_186: sensor.solarman_pv1_power
  pv2_power_187: sensor.solarman_pv2_power
  pv1_voltage_109: sensor.solarman_pv1_voltage
  pv1_current_110: sensor.solarman_pv1_current
  pv2_voltage_111: sensor.solarman_pv2_voltage
  pv2_current_112: sensor.solarman_pv2_current
  total_pv_generation: sensor.energy_production_today
  remaining_solar: sensor.energy_production_today_remaining
  environment_temp: sensor.casa_temperatura_esterna
  battery_temp_182: sensor.solarman_battery_1_temperature
  inverter_voltage_L2: sensor.solarman_voltage_phase_s
  inverter_voltage_L3: sensor.solarman_voltage_phase_t
  inverter_current_L2: sensor.solarman_current_output_s
  inverter_current_L3: sensor.solarman_current_output_t
  radiator_temp_91: sensor.solarman_radiator_temperature_1
  essential_load1: sensor.macchina_total_istant
  load_power_L1: sensor.phase_a_power
  load_power_L2: sensor.phase_b_power
  load_power_L3: sensor.phase_c_power
  grid_ct_power_L2: sensor.phase_b_power
  grid_ct_power_L3: sensor.phase_c_power
inverter:
  three_phase: true
  model: goodwe_gridmode
  modern: true
  autarky: 'no'
  auto_scale: true
panel_mode: true
show_battery: true
show_grid: true
dynamic_line_width: true
large_font: true
decimal_places: 1`

if you see the power use by the house it lacks 300 watt, that difference is displayed correctly in the other card. Both cards is using the same sensor.
Any ideas?

ps. i don't have a goodwe inverter but a zcs by zucchetti.

**Comment by slipx06** [comment]
Hello. Since you don't have a sensor defined for `essential_power` the card is trying to calculate it for you based on the formula below

` essential =  load_power_L1 +  load_power_L2 +  load_power_L3`

Which in your case is `sensor.phase_a_power  + sensor.phase_b_power + sensor.phase_c_power `

What value do you get when you add these manually?

**Reply by nebbiadigiorno** [reply]
i get the value display in the card, that is the same from the grid and the load. In the ha forum you tell me not to use the essential sensor but to use the 3 phase sensors. The card should consider the solar production, what goes to the battery and what goes to the house. 
This sensor is the same i am using in the other card, that display the value correctly

**Reply by slipx06** [reply]
Can you show us your sensor values  for the Grid and Load from HA developer tools?

**Reply by nebbiadigiorno** [reply]
the grid and the load has the same sensor, and in the screenshots are the "fase a,b,c"  in the central card. What you want to see from the developer tools?

**Comment by nebbiadigiorno** [comment]
<img width="857" alt="Screenshot 2024-02-19 alle 15 05 36" src="https://github.com/slipx06/sunsynk-power-flow-card/assets/59627032/cb611070-1a9b-4b09-b015-79dc5fc4582d">
<img width="857" alt="Screenshot 2024-02-19 alle 15 05 57" src="https://github.com/slipx06/sunsynk-power-flow-card/assets/59627032/0f200cfb-f466-46d4-9080-cf81ee73d10b">
<img width="857" alt="Screenshot 2024-02-19 alle 15 06 08" src="https://github.com/slipx06/sunsynk-power-flow-card/assets/59627032/9d058abf-4927-4a3b-bfab-ed3cd2fadbd2">

**Reply by slipx06** [reply]
Does your integration not provide you with sensor data for your load? 
The card wont work without this data? Technically your grid input cannot = your load output as you will always have losses (power to run inverter) or you may be using some of the grid to charge the battery and power the load etc

**Reply by nebbiadigiorno** [reply]
this are the sensors provided by my inverter integration:
<img width="264" alt="Screenshot 2024-02-19 alle 15 15 10" src="https://github.com/slipx06/sunsynk-power-flow-card/assets/59627032/64c91bad-8e5d-4477-8239-ef9831e87982">

**Reply by slipx06** [reply]
Can you provide a link to the integration you are using in HA

**Reply by nebbiadigiorno** [reply]
yes, [https://github.com/StephanJoubert/home_assistant_solarman](url)

**Reply by slipx06** [reply]
What inverter definition are you using? This one does not have many sensors https://github.com/StephanJoubert/home_assistant_solarman/blob/main/custom_components/solarman/inverter_definitions/zcs_azzurro-ktl-v3.yaml

**Reply by nebbiadigiorno** [reply]
i am using the sofar_g3hyd.yaml

**Comment by slipx06** [comment]
OK change your sensors (check the name) for the load to these and test:

```
load_power_L1: sensor.solarman_ActivePower_Load_R_EPS
load_power_L2: sensor.solarman_ActivePower_Load_S_EPS
load_power_L3: sensor.solarman_ActivePower_Load_T_EPS
```
and similarly change your grid sensors to these:


~~```~~
~~grid_ct_power_172: sensor.solarman_ActivePower_PCC_R~~
~~grid_ct_power_L2: sensor.solarman_ActivePower_PCC_S~~
~~grid_ct_power_L3: sensor.solarman_ActivePower_PCC_T~~

~~```~~
actually grid sensors should be

```
grid_ct_power_172: sensor.solarman_ActivePower_Output_R
grid_ct_power_L2: sensor.solarman_ActivePower_Output_S
grid_ct_power_L3: sensor.solarman_ActivePower_Output_T
```

**Reply by nebbiadigiorno** [reply]
nope the eps is not used in my system (the eps is the "ups" emergency system).
i used the active_power_load_sys in the essential power sensor and that does the trick

**Reply by slipx06** [reply]
Ok. remember to set three_phase: false
